plonky2x/core/src/frontend/uint/uint32.rs

```diff
@@ -207,6 +207,36 @@ impl U32Variable {
         }
         result
     }
+
+    pub fn from_be_bits<L: PlonkParameters<D>, const D: usize>(
+        bools: &[BoolVariable],
+        builder: &mut CircuitBuilder<L, D>,
+    ) -> Self {
+        assert!(bools.len() <= 32);
+        // We simply sum the bits and don't need to do a range-check, as we know that the sum of 32 bits is always less than 2^32
+        let var = builder.api.le_sum(
+            bools
+                .iter()
+                .rev()
+                .map(|b| (*b).into())
+                .collect::<Vec<BoolTarget>>()
+                .into_iter(),
+        );
+        Self(Variable(var))
+    }
+
+    pub fn to_be_bits<L: PlonkParameters<D>, const D: usize>(
+        &self,
+        builder: &mut CircuitBuilder<L, D>,
+    ) -> [BoolVariable; 32] {
+        let mut bits = builder.api.split_le(self.0 .0, 32);
+        bits.reverse();
+        bits.iter()
+            .map(|b| BoolVariable(Variable(b.target)))
```

Comment:
This can be changed to `BoolVariable::from(b)`.

Reply:
thanks